docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,806 @@
+
+The contents of the files we need to create have to be base64-encoded.
+
+We will create three files:
+
+**`/etc/hostname`**:
+
+```
+openshift-bootstrap
+```
+
+**`/etc/NetworkManager/conf.d/dhcp-client.conf`**:
+
+```
+[main]
+dhcp=dhclient
+```
+
+**`/etc/dhcp/dhclient.conf`**:
+
+```
+send dhcp-client-identifier = hardware;
+prepend domain-name-servers 127.0.0.1;
+```
+
+**NOTE**: We recommend you back up the Ignition files before making any changes!
+
+You can edit the Ignition file manually or run this Python script:
+
+```python
+import base64, json
+
+with open('bootstrap.ign', 'r') as f:
+    ignition = json.load(f)
+
+files = ignition['storage'].get('files', [])
+
+hostname_b64 = base64.encodestring('openshift-bootstrap\n').strip()
+files.append(
+{
+    'path': '/etc/hostname',
+    'mode': 420,
+    'contents': {
+        'source': 'data:text/plain;charset=utf-8;base64,' + hostname_b64,
+        'verification': {}
+    },
+    'filesystem': 'root',
+})
+
+dhcp_client_conf_b64 = base64.encodestring('[main]\ndhcp=dhclient\n').strip()
+files.append(
+{
+    'path': '/etc/NetworkManager/conf.d/dhcp-client.conf',
+    'mode': 420,
+    'contents': {
+        'source': 'data:text/plain;charset=utf-8;base64,' + dhcp_client_conf_b64,
+        'verification': {}
+        },
+    'filesystem': 'root',
+})
+
+dhclient_cont_b64 = base64.encodestring('send dhcp-client-identifier = hardware;\nprepend domain-name-servers 127.0.0.1;\n').strip()
+files.append(
+{
+    'path': '/etc/dhcp/dhclient.conf',
+    'mode': 420,
+    'contents': {
+        'source': 'data:text/plain;charset=utf-8;base64,' + dhclient_cont_b64,
+        'verification': {}
+        },
+    'filesystem': 'root'
+})
+
+ignition['storage']['files'] = files;
+
+with open('bootstrap.ign', 'w') as f:
+    json.dump(ignition, f)
+```
+
+Feel free to make any other changes.
+
+### Update Master Ignition
+
+Similar to bootstrap, we need to make sure the hostname is set to the expected value (it must match the name of the Nova server exactly).
+
+Since that value will be different for every master node, we will need to create multiple Ignition files: one for every node.
+
+We will deploy three Control plane (master) nodes. Their Ignition configs can be create like so:
+
+```sh
+$ for index in $(seq 0 2); do
+> MASTER_HOSTNAME="openshift-qlvwv-master-$index\n"
+> python -c "import base64, json, sys;
+ignition = json.load(sys.stdin);
+files = ignition['storage'].get('files', []);
+files.append({'path': '/etc/hostname', 'mode': 420, 'contents': {'source': 'data:text/plain;charset=utf-8;base64,' + base64.encodestring('$MASTER_HOSTNAME').strip(), 'verification': {}}, 'filesystem': 'root'});
+ignition['storage']['files'] = files;
+json.dump(ignition, sys.stdout)" <master.ign >"master-$index-ignition.json"
```

Comment:
Why don't we insert the random generated string (using @mandre 's "$infraID" suggestion) in the name of the master-$index-ignition.json file?  If the admin is setting up multiple clusters, wouldn't this be a more user friendly approach:  they would know which host the IGN file applies to and they don't risk clobbering a previous file?

Reply:
These ignition files are stored in a directory that does *not* support multiple deployments. You have to have a separate dir for each deployment.

That said, I've added this anyway to make things more consistent. It's nice to know which ignition to pass to which server since their names match.